services/wfm/src/main/java/org/openkilda/wfm/topology/isllatency/service/CacheService.java

```diff
@@ -0,0 +1,140 @@
+    private Map<Endpoint, Endpoint> cache;
+
+    public CacheService(CacheCarrier carrier, RepositoryFactory repositoryFactory) {
+        this.carrier = carrier;
+        islRepository = repositoryFactory.createIslRepository();
+        cache = new HashMap<>();
+        initCache();
+    }
+
+    private void initCache() {
+        cache.clear();
+        try {
+            islRepository.findAllActive()
+                    .forEach(isl -> {
+                        Endpoint source = Endpoint.of(isl.getSrcSwitch().getSwitchId(), isl.getSrcPort());
+                        Endpoint destination = Endpoint.of(isl.getDestSwitch().getSwitchId(), isl.getDestPort());
+
+                        cache.put(source, destination);
+                        cache.put(destination, source);
+                    });
+            log.debug("Isl latency cache: {}", cache);
+            log.info("Isl latency cache: Initialized");
+        } catch (Exception ex) {
+            log.error("Error during isl latency cache initialization", ex);
+        }
+    }
+
+    /**
+     * Remove outdated data from cache.
+     *
+     * @param notification notification about ISL status update
+     */
+    public void handleUpdateCache(IslStatusUpdateNotification notification) {
+        if (notification.getStatus() == IslStatus.MOVED || notification.getStatus() == IslStatus.INACTIVE) {
+            Endpoint source = Endpoint.of(notification.getSrcSwitchId(), notification.getSrcPortNo());
+            Endpoint destination = Endpoint.of(notification.getDstSwitchId(), notification.getDstPortNo());
+
+            cache.remove(source);
+            cache.remove(destination);
+
+            log.info("Remove ISL {}_{} ===> {}_{} from isl latency cache",
+                    notification.getSrcSwitchId(), notification.getSrcPortNo(),
+                    notification.getDstSwitchId(), notification.getDstPortNo());
+        }
+    }
+
+    /**
+     * Get data about specified ISL from cache.
+     *
+     * @param roundTripLatency round trip latency info data with source endpoint
+     */
+    public void handleGetDataFromCache(IslRoundTripLatency roundTripLatency) {
+        Endpoint source = Endpoint.of(roundTripLatency.getSrcSwitchId(), roundTripLatency.getSrcPortNo());
+
+        Endpoint destination = cache.get(source);
+
+        if (destination == null) {
+            destination = updateCache(source);
+        }
+
+        if (destination != null) {
+            carrier.emitCachedData(roundTripLatency, destination);
+        }
+    }
+
+    private Endpoint updateCache(Endpoint source) {
+        Isl isl = getActiveIsl(source);
+
+        if (isl == null) {
+            return null;
+        }
+
+        Endpoint destination = Endpoint.of(isl.getDestSwitch().getSwitchId(), isl.getDestPort());
+
+        cache.put(source, destination);
+        cache.put(destination, source);
+
+        return destination;
+    }
+
+    private Isl getActiveIsl(Endpoint source) {
```

Comment:
how often this part will be called?

Reply:
1 time for each ISL on topology start. and on each ISL UP event. 
not often